src/program/program.py

```diff
@@ -304,6 +272,8 @@ def _schedule_services(self) -> None:
             logger.debug(f"Scheduled {service_cls.__name__} to run every {update_interval} seconds.")
 
     def display_top_allocators(self, snapshot, key_type="lineno", limit=10):
+        import psutil
```

Comment:
I brought it here because I didnt want to add the additional lib import if tracemalloc wasnt enabled. 

Whats best in this scenario? Do we even care about pid memory?